src/client/sky.h

```diff
@@ -111,17 +112,24 @@ class Sky : public scene::ISceneNode
 		const std::string &use_sun_tint);
 	void setInClouds(bool clouds) { m_in_clouds = clouds; }
 	void clearSkyboxTextures() { m_sky_params.textures.clear(); }
-	void addTextureToSkybox(const  std::string &texture, int material_id,
+	void addTextureToSkybox(const std::string &texture, int material_id,
 		ITextureSource *tsrc);
 	const video::SColorf &getCurrentStarColor() const { return m_star_color; }
+
+	// Note: the sky class doesn't even use these values. It just stores them.
+	// These should be moved to a better place in the future.
```

Comment:
Where should they be moved?

Reply:
perhaps directly into `Client` or `Game`, not entirely sure

Reply:
wouldn't that be worse? both of these classes are already quite big

Reply:
Hey... We had a bunch of debates about where to put the server fog distance and fog start settings.

> the sky class doesn't even use these values

That commend does not make sense to me.  Maybe no code in Sky is using this, but it is currently a Sky feature.
"fog" could be a feature of the sky, or a feature of "lighting" or its own. We decided that the sky API is the right place, so on the client it ends up in `Sky`. That still makes sense to me.

Let's be careful where we put things and before we make any rash moves.

I'd prefer if we remove that comment until we have a firm decision about where is the right place.

Reply:
Alright.

I'll use this opportunity to air another grievance however:
We have `SkyboxParams`, we receive it, push it through a client event and it arrives in the game code.
Now instead of giving it to the sky class where the actual business happens the game code pulls apart the structure and calls a bunch of methods on the sky class to effect the necessary changes.
This results in me having to add a `setFoobar`/`getFoobar` pair to `Sky` for every new property even though the sky class literally has a `SkyboxParams` member internally. how does this make sense??

Reply:
I agree with that :)

We can file another PR to cleanup the API.
And while we're at it make sure we are happy with where I put volumetric_lighting. (It's part of the lighting API, but could be part of cloud, sky, or fog :) )